### ISSUE TITLE
Slowdown due to many spaces

### ISSUE THREAD

**Issue by jeff-hykin** [jeff-hykin/cpp-textmate-grammar#235]
```
#define a a                                                                    \
```
There is some pattern that is causing many spaces to be extremely slow. Probably some kind of backtracking. This looks like another form of #218 

I don't think there have been any changes to the #define patterns, so this might be an issue with one one of the patterns failing. When I debug I'm probably going to just start commenting out random patterns in `:root` to see what affects it. It seems for me that the C syntax is unaffected, so that is a hint at what patterns might be the cause.

**Comment by matter123** [comment]
Yeah, its the same as #218. function_definition is causing the slowness. Unfortunately, I'm not sure function definitions can be removed from the macro pattern.

`npm run perf -- ~/slow.cpp`
```
match failure
pattern name             average time    total time
source.cpp:2207          917.927000     4589.635
source.cpp:3380            0.050013        0.250
source.cpp:5292            0.012791        0.026
... snip ...

matched, not chosen
pattern name             average time    total time
source.cpp:663             0.333769        0.334
source.cpp:5793            0.039258        0.039

matched, chosen
pattern name             average time    total time
source.cpp:663             0.333769        0.334
source.cpp:5268            0.037544        0.038
source.cpp:5793            0.025654        0.026
source.cpp:5292            0.019325        0.019
```

**Comment by matter123** [comment]
prepending with `LookAheadToAvoid(@space)` cuts the performance issue in half (456ms avg 2280 total).

**Comment by jeff-hykin** [comment]
Yeah, I think it might be related to the interaction of qualified type inside the function definition.

**Comment by jeff-hykin** [comment]
I added some optimizations, but I'll keep this issue open until its cleaned up (readable/reusable). I did run the tests though and things seem to be good so I pushed a version

**Comment by jeff-hykin** [comment]
@d0ggie I think the issue is fixed with `v1.11.5`, I'm going to do further testing today to make sure.

**Comment by jeff-hykin** [comment]
With further testing everything seems to be working well, let me know if you have any more issues 👍
